arangod/RocksDBEngine/RocksDBSyncThread.cpp

```diff
@@ -184,3 +201,10 @@ void RocksDBSyncThread::run() {
     }
   }
 }
+
+void RocksDBSyncThread::notifySyncListeners(
+    rocksdb::SequenceNumber seq) noexcept {
+  for (auto& listener : _syncListeners) {
+    listener->onSync(seq);
+  }
```

Comment:
I know it may look like bad design - it leaves the door open for starving the SyncThread.
I wanted to go with the simplest approach here, as my goal is to keep RocksDB changes to a minimum. I expect these changes will no longer be needed (and reverted) once we move to a new WAL.
Having the SyncThread notify listeners in a scheduled task would require it to have access to the SCHEDULER. That's not necessarily something bad, but unless there are any other use-cases for it, I would rather avoid it.